### PR TITLE
Refactor Backend::Connection (move start test backend and logger, etc)

### PR DESCRIPTION
**Pull request by mdeniz**
Deep refactoring of the Backend::Connection class, moved the start_test_backend method and the logger methods  to their own classes. Moved all the test related stuff to the Backend::Test class. Stop using meta-programming for defining this class.

### CODE REVIEW

File: src/api/lib/backend/logger.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    end
+
+    def self.info(method, host, port, path, response, start_time)
+      time_delta = Time.now - start_time
+      now = Time.now.strftime "%Y%m%dT%H%M%S"
+      @backend_logger.info "#{now} #{method} #{host}:#{port}#{path} #{response.code} #{time_delta}"
+      @backend_time += time_delta
+      Rails.logger.debug "request took #{time_delta} #{@backend_time}"
+
+      return unless CONFIG['extended_backend_log']
```
Comment (hennevogel) [inline src/api/lib/backend/logger.rb]: Sounds like `self.debug` :)

File: src/api/lib/backend/test.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# API for accessing to the backend
+module Backend
+  class Test
+    @backend = nil
+
+    # Starts the test backend
+    def self.start(wait_for_scheduler = false)
```
Comment (hennevogel) [inline src/api/lib/backend/test.rb]: *nitpick*: I would rather use named parameters for this...

`Backend::Test.start(true)` vs. `Backend::Test.start(wait_for_scheduler: true)`

Comment (mschnitzer) [inline src/api/lib/backend/test.rb]: I'd also suggest that.

Comment (mdeniz) [inline src/api/lib/backend/test.rb]: Yes, sure, I was also thinking about something like that. I will change it. Thanks

File: src/api/lib/backend/logger.rb (hunk 1 above)
Comment (mdeniz) [inline src/api/lib/backend/logger.rb]: Yes, is kind of horrible... just moved the code, but we can re-refactor this later if you don't mind.

**Comment by mdeniz** [comment]
@hennevogel please re-review 😄

File: src/api/lib/backend/test.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# API for accessing to the backend
+module Backend
+  class Test
+    @backend = nil
+
+    # Starts the test backend
+    def self.start(options = {})
```
Comment (hennevogel) [inline src/api/lib/backend/test.rb]: A complete hash, you've learned from the master I guess. Why isn't it called opts?! ;-)

Comment (mdeniz) [inline src/api/lib/backend/test.rb]: I hate shortening variable names... :bowtie: 

> A complete hash, you've learned from the master I guess. 
Is that ironic? hahaha

**Pull request merged**